src/bm_sim/calculations.cpp

```diff
@@ -310,6 +310,12 @@ struct crcCCITT {
 
 struct cksum16 {
   uint16_t operator()(const char *buf, size_t len) const {
+    BMLOG_DEBUG("Printing Data: {}", len);
+#ifdef BM_LOG_DEBUG_ON
+    for (size_t i = 0; i < len; i++)
+      std::printf("%02X", buf[i]);
+#endif
+    BMLOG_DEBUG("\nEnd\n");
```

Comment:
I am sorry but I do not think this is a good addition: if the packet is 9,000 bytes and we compute the checksum over the whole payload, are you really going to print the whole buffer?

This is also irrelevant to the typical bmv2 user. Unless there is a bug in p4c or bmv2, then printing this buffer serves no real purpose and clutters the logs.